pkg/ebpf/bpf_client.go

```diff
@@ -734,43 +734,41 @@ func (l *bpfClient) computeMapEntriesFromEndpointRules(firewallRules []EbpfFirew
 		if !strings.Contains(string(firewallRule.IPCidr), "/") {
 			firewallRule.IPCidr += v1alpha1.NetworkAddress(l.hostMask)
 		}
-		//TODO - Just Purge both the entries and avoid these calls for every CIDR
-		if utils.IsCatchAllIPEntry(string(firewallRule.IPCidr)) {
-			continue
-		}
 
-		if len(firewallRule.L4Info) == 0 {
-			l.logger.Info("No L4 specified. Add Catch all entry: ", "CIDR: ", firewallRule.IPCidr)
-			l.addCatchAllL4Entry(&firewallRule)
-			l.logger.Info("Total L4 entries ", "count: ", len(firewallRule.L4Info))
-		}
-		if utils.IsNonHostCIDR(string(firewallRule.IPCidr)) {
-			if existingL4Info, ok := nonHostCIDRs[string(firewallRule.IPCidr)]; ok {
-				firewallRule.L4Info = append(firewallRule.L4Info, existingL4Info...)
-			}
-			nonHostCIDRs[string(firewallRule.IPCidr)] = firewallRule.L4Info
-		} else {
-			if existingL4Info, ok := ipCIDRs[string(firewallRule.IPCidr)]; ok {
-				firewallRule.L4Info = append(firewallRule.L4Info, existingL4Info...)
+		if !isCatchAllIPEntryPresent {
```

Comment:
Will skip all the entries if there is a catch All entry in the list of CIDRs but we only want to skip this for the catch All entry alone. `utils.IsCatchAllEntry` might be what we want to use here.

Reply:
Makes sense, will update it

Reply:
updated the PR..